src/lang/base/PrettyPrinters.ml

```diff
@@ -206,14 +206,13 @@ let rec pp_literal_simplified l =
             if String.is_empty a then t else a ^ " ; " ^ t
           ) ^ "]" in
       ("(Message " ^ items ^ ")")
-    | Map ((_, _), kv) ->
-      (* we don't print mtype as that's printed for every entry. *)
+    | Map ((kt, vt), kv) ->
       let items = "[" ^
         (Caml.Hashtbl.fold (fun k v a ->
           let t = "(" ^ (pp_literal_simplified k) ^ " => " ^ (pp_literal_simplified v) ^ ")" in
             if String.is_empty a then t else a ^ "; " ^ t
           ) kv "")  ^ "]" in
-      ("(Map " ^ items ^ ")")
+      ("(Map {" ^ pp_typ kt ^ " -> " ^ pp_typ vt ^ "} "  ^ items ^ ")")
```

Comment:
I don't see any value in printing the actual map type because that's already visible for each entry (as described in the comment above).

#174 added this function with the goal of replacing the, then existing, sexp based printing which was more precise, but less user friendly. The goal of this function was to provide simple, easily readable printing for literals. There is also a JSON version that is for parsing my machines.

Reply:
This is useful for empty maps. Otherwise tests can detect differences, but the pretty-printer's output is not helpful.

Reply:
I don't particularly mind that we print the map type, although I also don't see a big benefit in doing so.

However, I think we should try to stick to the type syntax of Scilla, and `t1 -> t2` is used only in connection with function types, not map types.